pinot-tools/src/main/java/org/apache/pinot/tools/service/PinotServiceManager.java

```diff
@@ -211,17 +206,20 @@ public boolean stopPinotInstance(ServiceStartable instance) {
   }
 
   public void start() {
-    LOGGER.info("Registering service status handler");
-    ServiceStatus.setServiceStatusCallback(_instanceId, new PinotServiceManagerStatusCallback(this));
-
-    if (_port < 0) {
+    if (_port <= 0) {
       LOGGER.info("Skip Starting Pinot Service Manager admin application");
     } else {
+      LOGGER.info("Registering service status handler");
+      ServiceStatus.setServiceStatusCallback(_instanceId, new PinotServiceManagerStatusCallback(this));
```

Comment:
+1 to putting this in `else` here. Why does the service manager ever need a status callback? Does it play any role in the component running after startup? The map based service status seems to return `STARTING` if there aren't any health checks registered anyway.

Reply:
PinotServiceManager can exposes APIs in `PinotServiceManagerAdminApiApplication` to dynamically create pinot components like controller/broker/server/minion if you give PinotSM a port.

Reply:
Users can start a PinotSM without bootstrap services, then later on start/stop services on it.
We need one healthCheck here to tell that this PinotServiceMananger is up.